src/Board.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <vector>
+
+class Board
+{
+    const int row_number_, col_number_;
```

Comment:
Please move col_number to the next line.
Do we want a snake_case naming convention? In C++ usually CamelCase is used, but I do not insist.